arelle/ModelObject.py

```diff
@@ -140,7 +140,8 @@ def objectId(self, refId: str = "") -> str:
         :param refId: A string to prefix the refId for uniqueless (such as to use in tags for tkinter)
         :type refId: str
         """
-        return "_{0}_{1}".format(refId, self.objectIndex)
+        index:Union[int,str] = self.objectIndex if hasattr(self, 'objectIndex') else 'None'
+        return "_{0}_{1}".format(refId, index)
```

Comment:
nit: we can use getattr to simplify this a bit:

```suggestion
        return f"_{refId}_{getattr(self, 'objectIndex', 'None')}"
```